README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
+transactionRequest.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionRequestCallback() {
+   override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+       // Do something
+   }
+})
+
+// The transaction consumer listen for the event
+transactionConsumption.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback() {
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+      // Do something
+  }
+})
+```
+
+> Note: You might want to listen for the event later after you got the `TransactionRequest` or `TransactionConsumption` object, but you might not want to pass the object around.
+In this case, it is possible to keep only the `socket_topic` string directly. Then you can alternatively listening for custom events by using `joinChannel` method of the `OMGSocketClient` instance.
+This implementation will provide the exactly same result as the implementation above ☝️. For example,
+
+```kotlin
+// The transaction requestor listen for the event 
+socketClient.joinChannel(SocketTopic(transactionRequestTopic), listener = object: SocketCustomEventCallback.TransactionRequestCallback(){
+  override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+      // Do something
+  }
+})
+
+// The transaction consumer listen for the event
+socketClient.joinChannel(SocketTopic(transactionRequestTopic), listener = object: SocketCustomEventCallback.TransactionConsumptionCallback() {
+    override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+        // Do something
+    }
+
+    override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+        // Do something
+    }
+})
+```
+
+> Be careful, this way ☝️, it is possible to pass a wrong callback without any compile error. If that happen you won't receive any event from the callback.
+
+## Stop listening for the custom event
+
+When you don't need to receive events anymore, you should call `stopListening(client: SocketClient)` for the corresponding `Listenable` object.
+This will leave the corresponding socket channel and close the connection if no other channel is active.
+
+For example,
+
+```kotlin
+transactionRequest.stopListening(socketClient)
```

Comment:
Is this automatic once a transaction has been consumed?

Reply:
The transaction can be consumed more many times depends on the `max_consumption` params https://github.com/omisego/ewallet/blob/develop/docs/guides/transaction_request_flow.md

So the user need to decide to stop it by themselves

Reply:
Maybe this could be automatic when the transaction was consumed the max number of times 🤔 
Not sure if you have everything necessary to do that

Reply:
Will implement it later. I think this PR should focus on making it **at least** works.